Morcoso_Daisy/Countries of the world/script.js

```diff
@@ -0,0 +1,78 @@
+const searchButton = document.getElementById("searchButton");
+const countryInput = document.getElementById("countryInput");
+const countryDetails = document.getElementById("country_details");
+const sameRegionCountries = document.getElementById("same_region_countries");
+
+searchButton.addEventListener("click", async () => {
+  const countryName = countryInput.value.trim();
+
+  if (!countryName) {
+    alert("Please enter a country name");
+    return;
+  }
+
+  try {
+    const response = await fetch(
+      `https://restcountries.com/v3.1/name/${countryName}`
+    );
+    if (!response.ok) {
+      throw new Error("Country not found");
+    }
+    const countryData = await response.json();
+    const country = countryData[0];
+
+    const details = `
+      <h2>Country Details - ${country.name.common}</h2>
+      <img src="${country.flags.svg}" alt="Flag of ${country.name.common}">
+      <p><strong>Capital:</strong> ${
+        country.capital ? country.capital[0] : "N/A"
+      } </p>
+      <p><strong>Region:</strong> ${country.region ? country.region :"N/A"}</p>
+      <p><strong>Languages:</strong> ${
+        country.languages ? Object.values(country.languages).join(", ") : "N/A"
+      }</p>
+      <p><strong>Population:</strong> ${
+        country.population ? country.population.toLocaleString() : "N/A"
+      }</p>
+      <p><strong>Area:</strong> ${
+        country.area
+          ? country.area.toLocaleString() + " square kilometers"
+          : "N/A"
+      } </p>
```

Comment:
Observe proper structuring of elements. You can nest the elements but you should not overlap them.

For reference, follow the guidelines:
https://docs.google.com/presentation/d/1iCoz17mBxZtIQ9PifAW0jjKsuv4xDTlx-TPQxdLySow/edit#slide=id.g29e6ce31160_0_5